File: src/main/java/com/mparticle/kits/SkyhookIntentService.java
package com.mparticle.kits;

import android.app.IntentService;
import android.content.Intent;

import com.skyhookwireless.accelerator.AcceleratorClient;
import com.skyhookwireless.accelerator.CampaignVenue;

public class SkyhookIntentService
    extends IntentService {

    public SkyhookIntentService() {
        super("com.mparticle.kits.SkyhookIntentService");
    }

    @Override
    protected void onHandleIntent(final Intent intent) {
        final CampaignVenue campaignVenue = AcceleratorClient.getTriggeringCampaignVenue(intent);
        if (campaignVenue != null) {
            switch (AcceleratorClient.getCampaignVenueTransition(intent)) {
                case CampaignVenue.CAMPAIGN_VENUE_TRANSITION_ENTER:
                    SkyhookLog.i("entered: " + campaignVenue);
                    break;
                case CampaignVenue.CAMPAIGN_VENUE_TRANSITION_EXIT:
                    SkyhookLog.i("exited: " + campaignVenue);
                    break;
                default:
                    SkyhookLog.e("unknown trigger type: " + campaignVenue);
            }
        } else {
            SkyhookLog.e("unknown intent type: " + intent);
        }
    }
}

File: src/main/kotlin/com/mparticle/kits/SkyhookIntentService.kt
package com.mparticle.kits

import com.mparticle.kits.SkyhookLog.i
import com.mparticle.kits.SkyhookLog.e
import android.app.IntentService
import android.content.Intent
import com.skyhookwireless.accelerator.CampaignVenue
import com.skyhookwireless.accelerator.AcceleratorClient
import com.mparticle.kits.SkyhookLog

class SkyhookIntentService : IntentService("com.mparticle.kits.SkyhookIntentService") {
    override fun onHandleIntent(intent: Intent?) {
        val campaignVenue = AcceleratorClient.getTriggeringCampaignVenue(intent)
        if (campaignVenue != null) {
            when (AcceleratorClient.getCampaignVenueTransition(intent)) {
                CampaignVenue.CAMPAIGN_VENUE_TRANSITION_ENTER -> i(
                    "entered: $campaignVenue"
                )
                CampaignVenue.CAMPAIGN_VENUE_TRANSITION_EXIT -> i("exited: $campaignVenue")
                else -> e("unknown trigger type: $campaignVenue")
            }
        } else if (AcceleratorClient.hasError(intent)) {
            e("error code: " + AcceleratorClient.getErrorCode(intent))
        } else {
            e("unknown intent type: $intent")
        }
    }
}